helloworld-rs/src/test/java/org/jboss/as/quickstarts/helloworldrs/BasicRuntimeIT.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.quickstarts.helloworld;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+/**
+ * The very basic runtime integration testing.
+ * @author Ashwin Mehendale
+ * @author emartins
+ */
+public class BasicRuntimeIT {
+
+    private static final String DEFAULT_SERVER_HOST = "http://localhost:8080/helloworld-rs";
+
+    private static String getServerHost() throws IOException, InterruptedException, URISyntaxException{
+    String serverHost = System.getenv("SERVER_HOST");
+        if (serverHost == null) {
+            serverHost = System.getProperty("server.host");
+        }
+        if (serverHost == null) {
+            serverHost = DEFAULT_SERVER_HOST;
+        }
+        return serverHost;
+    }
+
+    @Test
+    public void testHTTPEndpointIsAvailable() throws IOException, InterruptedException, URISyntaxException {
+        String serverHost = this.getServerHost();
+        final HttpRequest request = HttpRequest.newBuilder()
+                .uri(new URI(serverHost))
+                .GET()
+                .build();
+        final HttpClient client = HttpClient.newBuilder()
+                .followRedirects(HttpClient.Redirect.ALWAYS)
+                .connectTimeout(Duration.ofMinutes(1))
+                .build();
+        final HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        assertEquals(200, response.statusCode());
+    }
+
+    @Test
+    public void testMessage() throws IOException, InterruptedException, URISyntaxException {
+        String serverHost = this.getServerHost();
+        final HttpRequest request = HttpRequest.newBuilder()
+                .uri(new URI(serverHost+"/rest/HelloWorld"))
+                .GET()
+                .build();
+        final HttpClient client = HttpClient.newBuilder()
+                .followRedirects(HttpClient.Redirect.ALWAYS)
+                .connectTimeout(Duration.ofMinutes(1))
+                .build();
+        final HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        assertTrue(response.body().contains("Hello World!"));
+    }
```

Comment:
It seems a little strange to me to have two different tests here. Both should likely be asserting on the status code.

Also, is there a reason not to use the REST Client given this is a REST quickstart?

Reply:
+1

Reply:
Created a new class that uses a REST client: https://github.com/wildfly/quickstart/pull/953/files#diff-76464e0c59b997d4e308fd7c0f653e92733f182358fe8ef6f620f132347e10cd